client/src/store/historyStore/model/filtering.js

```diff
@@ -49,12 +58,18 @@ function equals(attribute, query = null, converter = null) {
 /**
  * Checks if a query value is part of the item value
  * @param {*} attribute of the content item
+ * @param {*} query parameter if the attribute does not match the server query key
+ * @param {*} converter if item attribute value has to be transformed e.g. to a date.
  */
-function contains(attribute, query = null) {
+function contains(attribute, query = null, converter = null) {
     return {
         attribute,
+        converter,
         query: query || `${attribute}-contains`,
         handler: (v, q) => {
+            if (converter) {
+                q = converter(q);
```

Comment:
Maybe we should convert `v = converter(v)` here, to make sure that they are always formatted in the same fashion.